vita3k/gui/src/profiles_manager_dialog.cpp

```diff
@@ -18,49 +18,60 @@
-    static const fs::path user_path{ fs::path(host.pref_path) / "ux0/user" };
+    const auto update_online_id = std::find_if(host.cfg.users.begin(), host.cfg.users.end(), [&](const auto &user) {
+        return user.second == online_id;
+    });
+    const auto user_path{ fs::path(host.pref_path) / "ux0/user" / host.io.user_id};
     static const auto BUTTON_SIZE = ImVec2(60.f, 0.f);
 
     // Add user
-    ImGui::InputTextWithHint("##online_id", "Write your online ID", &gui.online_id);
+    ImGui::InputTextWithHint("##online_id", "Write your online ID", &online_id);
     if (ImGui::IsItemHovered())
         ImGui::SetTooltip("Set your Online ID with a maximum of 16 characters.");
     ImGui::SameLine();
-    if (ImGui::Button("Add Online ID") && !gui.online_id.empty())
+    if (ImGui::Button("Add Online ID") && !online_id.empty())
         ImGui::OpenPopup("Add Online ID");
     ImGui::PushStyleColor(ImGuiCol_Text, GUI_COLOR_TEXT_MENUBAR_OPTIONS);
     if (ImGui::BeginPopupModal("Add Online ID", nullptr, ImGuiWindowFlags_AlwaysAutoResize)) {
         ImGui::PopStyleColor();
 
-        if (gui.online_id.length() > 16)
-            ImGui::TextColored(GUI_COLOR_TEXT, "Error: Need reduce to length for online ID.\nCurrently the length is: %d.", gui.online_id.length());
-        else if (update_online_id != host.cfg.online_id.end())
-            ImGui::TextColored(GUI_COLOR_TEXT, "Error: '%s' Online ID already exists.", gui.online_id.c_str());
+        if (online_id.length() > 16)
+            ImGui::TextColored(GUI_COLOR_TEXT, "Error: Need reduce to length for online ID.\nCurrently the length is: %lu.", online_id.length());
+        else if (update_online_id != host.cfg.users.end())
+            ImGui::TextColored(GUI_COLOR_TEXT, "Error: '%s' Online ID already exists.", online_id.c_str());
         else
-            ImGui::TextColored(GUI_COLOR_TEXT, "Success: '%s' Online ID added.", gui.online_id.c_str());
+            ImGui::TextColored(GUI_COLOR_TEXT, "Success: '%s' Online ID added.", online_id.c_str());
 
         ImGui::Separator();
         ImGui::SetCursorPosX(ImGui::GetWindowWidth() / 2 - 30);
         if (ImGui::Button("OK", BUTTON_SIZE)) {
-            if (update_online_id == host.cfg.online_id.end() && !(gui.online_id.length() > 16)) {
-                host.cfg.online_id.push_back(gui.online_id);
-                host.cfg.user_id += ((int)host.cfg.online_id.size() - 1) - host.cfg.user_id;
-                host.io.user_id = fmt::format("{:0>2d}", host.cfg.user_id);
-                if (!fs::exists(user_path / host.io.user_id))
-                    fs::create_directories(user_path / host.io.user_id / "savedata");
-                gui.online_id.clear();
+            if (update_online_id == host.cfg.users.end() && !(online_id.length() > 16)) {
+                auto id = 0;
+                for (auto u = host.cfg.users.begin(); u != host.cfg.users.end(); ++u, ++id)
```

Comment:
```suggestion
                for (auto user : host.cfg.users)
```

and then move `++id` into the equation?

Looks a bit cleaner than the `begin()/end()` imo
you did it right here :D https://github.com/Vita3K/Vita3K/pull/608/files#diff-5bafb201a20911edf33917a53a9fd63cR86

Reply:
Yes i know, just on this case i need incrément id value so is better like that

Reply:
if i using : i need write 
>if (id != u.first)
    break;
else
++id

so i have search for user other and get more small code, with increment id inside for